### ISSUE TITLE
Can't enroll new push token, invalid QR Code

### ISSUE THREAD

**Issue by Gelob** [privacyidea/privacyidea#1666]
# What did you try to do?
I configured cloud messaging in firebase and configured both enrollment and authentication policies in privacyidea. When I generate a new PUSH token I get the QR code and scan it with the privacyidea authentication app on android. The result is I get an invalid QR code message from the application. I thought maybe it was because I didn't configure a tokenlabel but even that didn't change anything. It seems like something is wrong with the URI in the QR code.

Please provide your top level intent!
Please provide steps to reproduce!

# What outcome did you expect?
The app to read the QR code and attempt to add the push token

What do you think, how the system should have worked?
It should have added it without issue

# What outcome did you experience?
Invalid QR code message in the app

What did actually happen?
Invalid QR code message in the app

# Configuration

* privacyIDEA Version: 3.0.1

* Installation method:
  (from source, github, virtualenv, ubuntu packages...)
from source/zip file

## more details: 

* OS: CentOS 7

* Webserver: nginx

* Tokendatabase: (like MySQL, PostgreSQL, ...)
MySQL

# Log file

**Turn on debug level and take a look at the privacyidea.log!**
**If appropriate, attach the log file.**

**Comment by cornelinux** [comment]
Did you use the privacyIDEA Authenticator from the beta test?
https://community.privacyidea.org/t/testing-privacyidea-push-token/1012

The Push token is not available in privacyIDEA Authenticator 1.x.

**Comment by Gelob** [comment]
Well now I feel stupid, that was it :P.

I'm having another issue as well. So now the token enrolled in the app and privacyidea says it enrolled successfully but when I test it using https://myserver/validate/check?user=username&pass=1234 I get back ERR401: Failed to submit message to firebase service.

privacyidea.log shows
```

  "error": {
    "code": 400,
    "message": "The registration token is not a valid FCM registration token",
    "status": "INVALID_ARGUMENT",
    "details": [
      {
        "@type": "type.googleapis.com/google.firebase.fcm.v1.FcmError",
        "errorCode": "INVALID_ARGUMENT"
      },
      {
        "@type": "type.googleapis.com/google.rpc.BadRequest",
        "fieldViolations": [
          {
            "field": "message.token",
            "description": "The registration token is not a valid FCM registration token"
          }
        ]
      }
    ]
  }
}
```

I see that when looking at the token in privacyidea this under info. `firebase_token: null`. I'm assuming that is the problem and its supposed to get populated when I enroll my token? Wondering if theirs a communication issue somewhere I'm missing or what. Sorry to tag onto this issue, figured it might be better than starting a new community thread atm.

**Comment by cornelinux** [comment]
Your firebase configuration is probably wrong!

You need to 
1. configure the firebase service
2. a policy for enrollment and
3. a policy for authentication.

**Comment by Gelob** [comment]
I did configure all 3 per that blog article and checked them for anything I'm missing per the screenshots/documentation. When should the firebase_token get populated? Is that upon enrollment of the token in the privacyidea application? Should the privacyidea server then contact firebase to add my phone as an endpoint?

**Comment by cornelinux** [comment]
@nilsbehlen can you take a look at this and we discuss this to make a plan?

**Comment by nilsbehlen** [comment]
Hi, did you use the beta version or the release one from the playstore?
In the beta there was the possibility that the firebase token was not received before continuing the rollout. That should be fixed in the release version (v2.0.1).

**Comment by Gelob** [comment]
It was 2.01 through the playstore. If its fixed in release then sounds good :)
